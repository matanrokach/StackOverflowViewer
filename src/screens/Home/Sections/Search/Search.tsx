import React from "react";
import { ErrorText, Title } from "../../../../components";
import { Button, Text, TextInput } from "../../../../elements";

interface Props {
	searchTerm: string;
	onSearchTermChange(searchTerm: string): void;
	onSubmit(): void;
	error: string;
}

export const Search = ({ searchTerm, onSearchTermChange, onSubmit, error }: Props) => {

	return (
		<>
			<Title>{'Enter user id:'}</Title>
				<TextInput
					{...{
						value: searchTerm,
						onChangeText: onSearchTermChange,
						placeholder: 'e.g: 123456',
					}}
				/>
				<Button onPress={onSubmit}>
					<Text>{'SEARCH'}</Text>
				</Button>

				<ErrorText {...{error}} />
		</>
	);
}