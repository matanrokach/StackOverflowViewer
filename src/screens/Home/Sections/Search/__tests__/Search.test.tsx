import {fireEvent, render} from '@testing-library/react-native';
import React from 'react';
import {Text, TouchableOpacity, View} from 'react-native';
import { Search } from '../Search';

describe('Search', () => {
  it('should match snapshot', () => {

		const props = {
			searchTerm: "some search term",
			onSearchTermChange: () => {},
			onSubmit: () => {},
			error: '',
		};

    const {toJSON} = render(<Search {...props} />);

		expect(toJSON()).toMatchSnapshot();

  });

	it('should have search term typed in text input', () => {

		const props = {
			searchTerm: "some search term",
			onSearchTermChange: () => {},
			onSubmit: () => {},
			error: '',
		};

		const { getByDisplayValue } = render(<Search {...props} />);

		expect(getByDisplayValue(props.searchTerm)).toBeDefined();

	});

	it('should be able to change search term', () => {

		const mockOnSearchTermChange = jest.fn();

		const props = {
			searchTerm: "some search term",
			onSearchTermChange: mockOnSearchTermChange,
			onSubmit: () => {},
			error: '',
		};

		const {getByDisplayValue} = render(<Search {...props} />);

		const textInput = getByDisplayValue(props.searchTerm);

		const newSearchTerm = 'new search term';

		fireEvent.changeText(textInput, newSearchTerm);

		expect(mockOnSearchTermChange).toHaveBeenCalledWith(newSearchTerm);

	});

	it('should have clickable search button', () => {

		const mockOnSubmit = jest.fn();

		const props = {
			searchTerm: "some search term",
			onSearchTermChange: () => {},
			onSubmit: mockOnSubmit,
			error: '',
		};

		const searchButtonText = "SEARCH";

		const {getByText} = render(<Search {...props} />);

		const button = getByText(searchButtonText);

		fireEvent.press(button);

		expect(mockOnSubmit).toHaveBeenCalled();

	});
	it('should have clickable search button', () => {

		const mockOnSubmit = jest.fn();

		const props = {
			searchTerm: "some search term",
			onSearchTermChange: () => {},
			onSubmit: mockOnSubmit,
			error: '',
		};

		const searchButtonText = "SEARCH";

		const {getByText} = render(<Search {...props} />);

		const button = getByText(searchButtonText);

		fireEvent.press(button);

		expect(mockOnSubmit).toHaveBeenCalled();

	});
	
	it('should show error', () => {

		const props = {
			searchTerm: "some search term",
			onSearchTermChange: () => {},
			onSubmit: () => {},
			error: 'some error',
		};

		const {getByText} = render(<Search {...props} />);

		const errorText = getByText(props.error);

		expect(errorText).toBeDefined();

	});
	
});
