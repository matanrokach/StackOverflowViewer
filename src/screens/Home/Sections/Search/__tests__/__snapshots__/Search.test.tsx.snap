// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should match snapshot 1`] = `
Array [
  <Text
    style={
      Object {
        "color": "rgb(28, 28, 30)",
        "fontSize": 20,
        "fontWeight": "500",
      }
    }
  >
    Enter user id:
  </Text>,
  <View
    style={
      Array [
        Object {
          "alignSelf": "stretch",
          "borderRadius": 6,
          "borderWidth": 1,
          "marginHorizontal": 20,
          "marginVertical": 10,
          "paddingHorizontal": 10,
        },
        Object {
          "backgroundColor": "rgb(255, 255, 255)",
        },
      ]
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="e.g: 123456"
      style={
        Object {
          "backgroundColor": "rgb(255, 255, 255)",
          "borderColor": "rgb(216, 216, 216)",
          "paddingVertical": 10,
        }
      }
      value="some search term"
    />
  </View>,
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#292828",
        "borderColor": "#efefef",
        "borderRadius": 6,
        "borderWidth": 1,
        "opacity": 1,
        "paddingHorizontal": 10,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "rgb(28, 28, 30)",
        }
      }
    >
      SEARCH
    </Text>
  </View>,
  <View
    style={
      Object {
        "height": 24,
        "paddingVertical": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#cc0000",
        }
      }
    >
      
    </Text>
  </View>,
]
`;
